File: tzapi4android/src/main/java/com/tianzunchina/android/api/utils/PhoneTools.java
package com.tianzunchina.android.api.utils;

import android.app.Activity;
import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.net.Uri;
import android.provider.Settings;
import android.telephony.TelephonyManager;

import com.tianzunchina.android.api.control.TZApplication;

import java.io.File;

/**
 * 手机信息工具类
 * CraetTime 2016-3-4
 * @author SunLiang
 */
public class PhoneTools {
    TZApplication app = TZApplication.getInstence();
    TelephonyManager tm = (TelephonyManager) app.getSystemService(Context.TELEPHONY_SERVICE);
    PackageManager pm = app.getPackageManager();

    /**
     * 应用版本号
     * @return
     */
    public int getVersionCode() {
        int version = 0;
        try {
            PackageInfo packInfo = pm.getPackageInfo(app.getPackageName(), 0);
            version = packInfo.versionCode;
        } catch (Exception e ) {
            e.printStackTrace();
        }
        return version;
    }

    /**
     * 应用版本名称
     * @return
     */
    public String getVersionName(){
        String versionName = null;
        try {
            PackageInfo packInfo = pm.getPackageInfo(app.getPackageName(), 0);
            versionName = packInfo.versionName;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return versionName;
    }

    /**
     * 唯一的设备ID： GSM手机的 IMEI 和 CDMA手机的 MEID.
     * @return 获得设备ID
     */
    String getDeviceId(){
        return tm.getDeviceId();
    }

    /**
     * 手机号码
     * @return 获取本机号码
     */
    String getPhoneNumber(){
        return tm.getLine1Number();
    }


    /**
     * 打开系统设置
     * @param context
     */
    public void showSettingDate(Context context ) {
        context.startActivity(new Intent(Settings.ACTION_DATE_SETTINGS));
    }

    /**
     * 安装APK(获取意图)
     * @param path .apk文件路径
     * @return
     */
    public Intent getApkFileIntent(String path)  {
        Intent intent = new Intent();
        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        intent.setAction(android.content.Intent.ACTION_VIEW);
        Uri uri = Uri.fromFile(new File(path));
        intent.setDataAndType(uri, "application/vnd.android.package-archive");
        return intent;
    }

    public Intent getImageFileIntent(File file) {
        Uri uri = Uri.fromFile(file);
        return getImageFileIntent(uri);
    }

    /**
     * 通过系统图片浏览器打开图片(获取意图)
     * @param uri
     * @return
     */
    public Intent getImageFileIntent(Uri uri) {
        Intent intent = new Intent(Intent.ACTION_VIEW);
        intent.addCategory(Intent.CATEGORY_DEFAULT);
        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        intent.setDataAndType(uri, "image/*");
        return intent;
    }

    /**
     * 打开时间设置页面
     * @param activity
     */
    public void  openDatetimeSetting(Activity activity ) {
        openSetting(activity, SETTING_DATETIME);
    }

    /**
     * 打开GPS定位设置页面
     * @param activity
     */
    public void  openGPSSetting(Activity activity ) {
        openSetting(activity, SETTING_GPS);
    }

    /**
     * 打开设置界面
     */
    public void openSetting(Activity activity , String setting) {
        Intent intent = new Intent("/");
        try {
            ComponentName cm = new ComponentName("com.android.settings",
                    "com.android.settings." + setting);
            intent.setComponent(cm);
            intent.setAction(Intent.ACTION_VIEW);
            activity.startActivityForResult(intent, 0);
        } catch (Exception e) {
        }
    }

    private static final String SETTING_DATETIME = "DateTimeSettings";
    private static final String  SETTING_GPS = "SecuritySettings";
}

File: tzapi4android/src/main/java/com/tianzunchina/android/api/utils/PhoneTools.kt
package com.tianzunchina.android.api.utils

import android.app.Application
import android.content.Context
import android.content.Intent
import android.content.pm.PackageInfo
import android.content.pm.PackageManager
import android.net.Uri
import android.provider.Settings
import android.telephony.TelephonyManager
import com.tianzunchina.android.api.control.TZApplication

import java.io.File

/**
 * 手机信息工具类
 * CraetTime 2016-3-4
 * @author SunLiang
 */
object  PhoneTools {
	var tm: TelephonyManager
	var pm: PackageManager
	val app = TZApplication.instance

    init {
        tm = app!!.getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
        pm = app.packageManager
    }

    val versionCode: Int
        get() {
            var packInfo: PackageInfo?
            var version = 0
            try {
                packInfo = pm.getPackageInfo(app!!.packageName, 0)
                version = packInfo!!.versionCode
            } catch (e: Exception) {
                e.printStackTrace()
            }

            return version
        }

    val versionName: String
        get() {
            var packInfo: PackageInfo?
            var versionName = ""
            try {
                packInfo = pm.getPackageInfo(app!!.packageName, 0)
                versionName = packInfo!!.versionName
            } catch (e: Exception) {
                e.printStackTrace()
            }

            return versionName
        }

    /**
	 * 唯一的设备ID： GSM手机的 IMEI 和 CDMA手机的 MEID.
	 * @return 获得设备ID
	 */
    val deviceId: String
        get() = tm.deviceId

    /**
	 * 手机号码
	 * @return 获取本机号码
	 */
    val line1Number: String
        get() = tm.line1Number

    val simCountryIso: String
        get() = tm.simCountryIso

    fun showSettingDate(context: Context) {
        context.startActivity(Intent(Settings.ACTION_DATE_SETTINGS))
    }

	fun getApkFileIntent(param: String): Intent {
		val intent = Intent()
		intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
		intent.action = android.content.Intent.ACTION_VIEW
		val uri = Uri.fromFile(File(param))
		intent.setDataAndType(uri, "application/vnd.android.package-archive")
		return intent
	}

	fun getImageFileIntent(file: File): Intent {
		val uri = Uri.fromFile(file)
		return getImageFileIntent(uri)
	}

	fun getImageFileIntent(uri: Uri): Intent {
		val intent = Intent("android.intent.action.VIEW")
		intent.addCategory("android.intent.category.DEFAULT")
		intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
		intent.setDataAndType(uri, "image/*")
		return intent
	}
}
